libsolidity/formal/SMTEncoder.cpp

```diff
@@ -883,6 +886,102 @@ void SMTEncoder::endVisit(ElementaryTypeNameExpression const& _typeName)
 	m_context.createExpression(_typeName, result.second);
 }
 
+namespace
+{
+unsigned dimensionCount(TypePointer _type)
+{
+	unsigned dimensionCount = 0;
+	while (_type->category() == Type::Category::Array || _type->category() == Type::Category::Mapping)
+	{
+		++dimensionCount;
+		if (_type->category() == Type::Category::Array)
+			_type = dynamic_cast<ArrayType const *>(_type)->baseType();
+		else if (_type->category() == Type::Category::Mapping)
+			_type = dynamic_cast<MappingType const *>(_type)->valueType();
+		else
+			solAssert(false, "");
+	}
+	return dimensionCount;
+}
+
+vector<string> structGetterReturnedMembers(StructType const* _structType)
+{
+	vector<string> returnedMembers;
+	for (auto const& member: _structType->nativeMembers(nullptr))
+	{
+		// TODO: Figure out which types are returned by the getter and which are ignored
+		if (auto category = member.type->category(); category == Type::Category::Array || category == Type::Category::Mapping)
+			continue;
+		returnedMembers.push_back(member.name);
+	}
+	return returnedMembers;
+}
+
+}
+
+void SMTEncoder::visitPublicGetter(FunctionCall const& _funCall)
+{
+	MemberAccess const& access = dynamic_cast<MemberAccess const&>(_funCall.expression());
+	auto var = dynamic_cast<VariableDeclaration const*>(access.annotation().referencedDeclaration);
+	solAssert(var, "");
+	solAssert(m_context.knownExpression(_funCall), "");
+	auto actualArguments = applyMap(_funCall.arguments(), [this](auto const& argExpr) { return expr(*argExpr); });
```

Comment:
```suggestion
	auto actualArguments = applyMap(_funCall.arguments(), [this](auto argExpr) { return expr(*argExpr); });
```
I think it's good practice to not use & to smart pointer

Reply:
Fixed.